spec/requests/requests_spec.rb

```diff
@@ -6,20 +6,28 @@
   let!(:template) { create(:template) }
   let!(:workflow) { create(:workflow, template_id: template.id) }
   let(:workflow_id) { workflow.id }
-  let!(:requests) { create_list(:request, 20, workflow_id: workflow.id, status: 'PENDING', state: 'QUEUED') }
+  let!(:requests) { create_list(:request, 2, workflow_id: workflow.id) }
   let(:id) { requests.first.id }
+  let(:req_uuid) { requests.first.uuid }
+  let!(:requests_with_same_uuid) { create_list(:request, 2, uuid: 'abcd', workflow_id: workflow.id) }
+  let!(:requests_with_same_state) { create_list(:request, 2, state: 'notified', workflow_id: workflow.id) }
+  let!(:requests_with_same_status) { create_list(:request, 2, status: 'approved', workflow_id: workflow.id) }
+
+
+  let(:user_encode_key) { { 'x-rh-auth-identity': 'eyJpZGVudGl0eSI6eyJpc19vcmdfYWRtaW4iOmZhbHNlfX0=\n' } }
+  let(:admin_encode_key) { { 'x-rh-auth-identity': 'eyJpZGVudGl0eSI6eyJpc19vcmdfYWRtaW4iOnRydWV9fQ==\n' } }
 
   # Test suite for GET /workflows/:workflow_id/requests
   describe 'GET /workflows/:workflow_id/requests' do
-    before { get "/workflows/#{workflow_id}/requests" }
+    before { get "/workflows/#{workflow_id}/requests", headers: admin_encode_key }
```

Comment:
@hsong-rh - is line working for you? I'm not able to find key in the request headers by simply looking it up by 'x-rh-auth-identity' Though I can find it named `HTTP_X_RH_AUTH_IDENTITY` in the request object.  Wondering if you are seeing the same behavior?